### ISSUE TITLE
WCS axes projection gives error when tick direction is "inout"

### ISSUE THREAD

**Issue by ysBach** [astropy/astropy#12339]
### Description

WCS axes projection gives error when tick direction is "inout"

### Expected behavior

Set `ax.tick_params(direction="inout")`

### Actual behavior

```
ValueError: direction should be 'in' or 'out'
```

### Steps to Reproduce

```python
from astropy.coordinates import SkyCoord
from astropy.wcs import WCS
hdu = SkyView.get_images(SkyCoord(ra=0*u.deg, dec=0*u.deg), survey="DSS", pixels=10, cache=False)[0][0]
wcs = WCS(hdu.header)
fig = plt.figure(figsize=(6, 6))
ax = plt.subplot(projection=wcs)
ax.imshow(hdu.data, origin='lower')
ax.tick_params(direction="inout")
plt.show();
```

### System Details

```
macOS-10.16-x86_64-i386-64bit
Python 3.8.12 (default, Oct 12 2021, 06:23:56) 
[Clang 10.0.0 ]
Numpy 1.21.2
pyerfa 2.0.0
astropy 4.3.1
Scipy 1.7.1
Matplotlib 3.4.3
```

As astropy requires `matplotlib !=3.4.0,>=3.1 or later` and those matplotlib versions support all three of `('in', 'out', 'inout')`, the code may be updated as follows:
```python
# Original code: https://github.com/astropy/astropy/blob/main/astropy/visualization/wcsaxes/coordinate_helpers.py#L359
        if direction is not None:
            if direction in ('in', 'out'):
                self.ticks.set_tick_out(direction == 'out')
            else:
                raise ValueError("direction should be 'in' or 'out'")

# Fix to
        if direction is not None:
            if direction in ('in', 'out', 'inout'):
                self.ticks.set_tick_out(direction == 'out')
            else:
                raise ValueError("direction should be 'in', 'out', or 'inout'")

```

**Comment by neutrinoceros** [comment]
Here's a more self-contained reproducer:
```python
import matplotlib.pyplot as plt
from astropy.wcs import WCS

fig = plt.figure()
ax = fig.add_axes([0.1, 0.1, 0.8, 0.8], projection=WCS())
ax.tick_params(direction="inout")
```

I note that the proposed fix isn't sufficient: it *does* avoid the exception, but it doesn't implement the correct behaviour. Currently, `astropy.visualization.wcsaxes.ticks.Ticks` only supports drawing a tick in *or* out, but not both. I see two things that could be done here:
- raise a warning when receiving `direction="inout"`, but do not try to enforce it (this would avoid the exception, but wouldn't solve the actual problem)
- replace `Ticks.set_tick_out` (which isn't extensible since it falls in the boolean trap) with a new API.

I think the second option is more desirable but I'm not expert enough in this part of the code to confidently estimate how long it'd take to implement it. In any case, I think this might be re-qualified as a feature request. What do you think @pllim ?

**Comment by pllim** [comment]
Thanks for the ping but I am not comfortable making that call. I'll defer to @astrofrog , @Cadair , or @larrybradley on this.

**Comment by Cadair** [comment]
Yes, I think this is a feature request. I think we could add support for it, but in the absence of that the error is the right thing to happen.

As for how long it would take, looks like it would be non-trivial. Easiest option might be to draw both in and out ticks (i.e. two ticks per position).
